library/src/main/scala/scala/scalajs/js/JSON.scala

```diff
@@ -46,6 +46,26 @@ object JSON extends js.Object {
    * is specified, or optionally including only the specified properties if a
    * replacer array is specified.
    *
+   *   - If the value has a
+   *     [[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior toJSON()]]
+   *     method, it's responsible to define what data will be serialized.
+   *   - [[Boolean]], [[Double Number]], and [[java.lang.String String]] objects are converted to
+   *     the corresponding primitive values during stringification, in accord
+   *     with the traditional conversion semantics.
+   *   - [[undefined]], [[Function]]s, and [[Symbol]]s are not valid JSON
+   *     values. If any such values are encountered during conversion they are
+   *     either omitted (when found in an [[Object]]) or changed to null (when
+   *     found in an [[Array]]).
+   *   - All [[Symbol]]-keyed properties will be completely ignored, even when
+   *     using the replacer function.
+   *   - The instances of [[Date]] implement the [[Date#toJSON()*]] function by
+   *     returning a string (the same as [[Date#toISOString]]). Thus, they are
+   *     treated as strings.
+   *   - The numbers Infinity and NaN, as well as the value null, are all
+   *     considered null.
+   *   - All the other [[Object]] instances (including Map, Set, WeakMap, and
+   *     WeakSet) will have only their enumerable properties serialized.
```

Comment:
We should add a paragraph that explains what happens when the top-level argument is a function or `undefined`. Basically a replacement for the MDN doc that says:

> JSON.stringify() can return undefined when passing in "pure" values like JSON.stringify(function(){}) or JSON.stringify(undefined).

In our case we should say that those calls are invalid and will result in Undefined Behavior.